tests/operators/test_autograd_linop.py

```diff
@@ -0,0 +1,91 @@
+"""Test for the adjoint as backward option in the LinearOperator class."""
+
+import pytest
+import torch
+from mrpro.operators import LinearOperator
+from torch.autograd.gradcheck import GradcheckError
+
+from tests import RandomGenerator
+from tests.helper import dotproduct_adjointness_test
+
+
+class NonDiffentiableOperator(LinearOperator, adjoint_as_backward=False):
+    """Dummy operator for testing purposes."""
+
+    def forward(self, x):
+        python_float = x.item()
+        tensor = torch.tensor([2 * python_float], dtype=x.dtype)  # breaks the autograd
+        return (tensor,)
+
+    def adjoint(self, x):
+        python_float = x.item()
+        tensor = torch.tensor([2 * python_float], dtype=x.dtype)  # breaks the autograd
+        return (tensor,)
+
+
+class DiffentiableOperator(LinearOperator, adjoint_as_backward=True):
+    """Dummy operator for testing purposes. Should be differentiable due to adjoint_as_backward=True."""
+
+    def forward(self, x):
+        python_float = x.item()
+        tensor = torch.tensor([2 * python_float], dtype=x.dtype)  # breaks the autograd
+        return (tensor,)
+
+    def adjoint(self, x):
+        python_float = x.item()
+        tensor = torch.tensor([2 * python_float], dtype=x.dtype)  # breaks the autograd
+        return (tensor,)
+
+
+def test_linop_autograd_dummy_operator():
+    """Ensure correct dummy operator."""
+    rng = RandomGenerator(seed=0)
+    nondiff = NonDiffentiableOperator()
+    diff = DiffentiableOperator()
+
+    u = rng.float32_tensor((1,))
+    v = rng.float32_tensor((1,))
+
+    # Differentiable operator and non differentiable operator should be the same
+    torch.testing.assert_close(nondiff.adjoint(u), nondiff.adjoint(u))
+    torch.testing.assert_close(diff.adjoint(u), diff.adjoint(u))
```

Comment:
not sure what is tested here. did you want to compare the forward of diff and nondiff?